### ISSUE TITLE
Accessibility: Improving the accessibility in enforcement guidelines and principles page

### ISSUE THREAD

**Issue by preshpi** [jupyterhub/outreachy#116]
### What page is this for?

https://github.com/jupyterhub/outreachy/issues/38#issuecomment-1286039376

### WAVE accessibility report

My name is Precious. I'm contributing on the [Jupyter.org](https://jupyter.org/governance/conduct/enforcement.html#enforcement-guidelines-and-principles).
<img width="1440" alt="Screen Shot 2022-10-20 at 4 56 47 PM" src="https://user-images.githubusercontent.com/80992383/197395336-2a76c91e-0e79-48a2-9627-8a444972d00a.png">
After evaluating this page with the wave tools, these were the errors I got.
https://wave.webaim.org/report#/https://jupyter.org/governance/conduct/enforcement.html#conflicts-of-interest

<h1> 9 Errors </h1>

- 5x Empty form label (A form label is present, but does not contain any content.)
  <b>SOLUTIONS:</b> Text should be included in the form label describing the form control's function.

- 2x Multiple-form labels (A form control has more than one label associated with it.)
  <b>SOLUTION:</b> Form controls should have no more than one label element associated with them. If multiple form labels are necessary, use `aria-labelledby`.

- 1x Language missing or invalid (The document's language is not identified or a lang attribute value is invalid.)
<b>SOLUTION:</b> Identify the document language using the <html lang> attribute with a valid value `<html lang="en">`. Ensure that all lang attribute values are valid.

- 1x Empty button (A button is empty or has no value text.)
<b>SOLUTION:</b> Place text content within the <button> element or give the <input> element a value attribute.

<h1> 5 Alerts </h1>

- 5x Possible heading (Text appears to be a heading but is not a heading element.) Heading elements `(<h1>-<h6>)` provide important document structure, outlines, and navigation functionality to assistive technology users. 
<b>SOLUTION:</b> If the paragraph is a section heading, use a heading element instead `(<h1>-<h6>).`

<h1> Features </h1>

- 1 X Linked image with an alternative text (Including appropriate alternative text on an image within a link ensures that the function and purpose of the link and the content of the image is available to screen reader users or when images are unavailable.)
<b> SOLUTION: </b>Ensure that the alternative text presents the content of the image and/or the function of the link. 

- 2 X Form label (A properly associated form label is presented to a screen reader user when the form control is accessed.)
<b>SOLUTION: </b> Ensure that the label is accurate, descriptive, succinct and that it is associated with the correct form control element.

<h1> 33 Structural Elements </h1>

- 3x Heading level 1 (Headings facilitate page navigation for users of assistive technologies.)
<b>SOLUTION: </b> Be sure that the text is truly a `<h1>` and is structured correctly on the page

- 7x Heading level 2 (Headings facilitate page navigation for users of assistive technologies)
<b>SOLUTION: </b> Be sure that the text is truly a `<h2>` and is structured correctly on the page

- 17x Unordered list  (Ordered lists present a group of related, parallel items. Users of many assistive technologies can navigate by and within lists.)
<b>SOLUTION:</b>Ensure that an unordered (bulleted) list is appropriate for the context. If list items are sequential or numbered, an ordered `<ol>` is likely more appropriate.

- 3 X Navigation
<b>SOLUTION:</b> Multiple navigation elements on one page can be differentiated with ARIA labels.

- 1x Main content
<b>SOLUTION:</b> The `<main>` element or `role="main"` attribute identifies the main content for the page.`main` should surround the page

- 2x Footer ( It typically identifies authorship, related links, copyright date, or other footer content. Footers facilitate page semantics and navigation.)
<b>SOLUTION: </b> `footer` should surround the footer content of the page

<h1>8 ARIA </h1>

- 1x ARIA (ARIA provides enhanced semantics and accessibility for web content.) An ARIA `role,` `state,` or `property` is present, excluding landmark roles, `aria-labelledby,` or `aria-describedby` which are distinct WAVE items.
<b>SOLUTION:</b> ARIA must be used correctly and also know that support for ARIA is limited in older browsers and assistive technologies.

 - 7x ARIA Label
<b>SOLUTION:</b> ARIA labels are important for screen readers which access names to be read by the screen reader. Ensure the `aria-label` value or element referenced by aria-labelledby provides a descriptive, accurate label. When possible, use standard HTML `<label>` or other markup to make the association.
 

### The HTML element to be changed

```html
9 ERRORS
1. Language missing or invalid

2. Empty form label
<label for="toctree-checkbox-1">
<i class="fas fa-chevron-down"></i>
</label>

3. Multiple-form labels
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">

4.  Empty button
<button onclick="toggleFullScreen()" class="headerbtn" data-toggle="tooltip" data-placement="bottom" title="" data-original-title="Fullscreen mode">
<span class="headerbtn__icon-container">
<i class="fas fa-expand"></i>
</span>
</button>
 
5 ALERTS

1. Possible heading
<p aria-level="2" class="caption" role="heading">
<span class="caption-text">
Code of Conduct
</span>
</p>

3 FEATURES
1. Linked image with an alternative text
<img src="https://licensebuttons.net/p/zero/1.0/88x31.png" alt="CC0" style="border-style: none;">

2.  Form label
<label class="overlay overlay-pagetoc" for="__page-toc">
<div class="visually-hidden">
Toggle in-page Table of Contents
</div>
</label>


Structural Elements
1. Unordered list
<ul class="nav bd-sidenav bd-sidenav__home-link">
<li class="toctree-l1">
<a class="reference internal" href="../intro.html">
Project Jupyter Governance
</a>
</li>
</ul>

2. Navigation
<nav aria-label="Page">
Unordered list
<ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#the-code-of-conduct-committee">
The Code of Conduct Committee
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#enforcement-guidelines-and-principles">
Enforcement guidelines and principles
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#how-the-committee-will-respond-to-reports">
How the committee will respond to reports
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#incident-response-and-committee-actions">
Incident Response and Committee Actions
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#resolutions">
Resolutions
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#conflicts-of-interest">
Conflicts of Interest
</a>
</li>
</ul>
</nav>

8 ARIA

1. ARIA
<p aria-level="2" class="caption" role="heading">
<span class="caption-text">
Code of Conduct
</span>
</p>

2. ARIA-LABEL

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">

<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
```

### Your proposed HTML (or CSS), after change

```html
1. Language missing or invalid:
 I suggest that languages should be stated <html lang="en"> 

2.Empty form label
The form label contains text that describes the function of the associated form control. 
<label for="toctree-checkbox-1">
Arrow down
<i class="fas fa-chevron-down"></i>
</label>

3. Multiple form labels: If multiple form labels are necessary, use aria-labelledby.
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-labelledby="Toggle navigation sidebar">

4.  Empty button
Place text content within the <button> element or give the <input> element a value attribute.

<button onclick="toggleFullScreen()" class="headerbtn" data-toggle="tooltip" data-placement="bottom" title="" data-original-title="Fullscreen mode">
<span class="headerbtn__icon-container">
<i class="fas fa-expand"></i>
</span>
Button text
</button>

5 ALERTS

1. Possible heading
If the paragraph is a section heading, use a heading element instead (<h1>-<h6>).

<h2 class="caption">
<span class="caption-text">
Code of Conduct
</span>
</h2>

3 FEATURES
1. Linked image with alternative text
The alt text should be descriptive. I'm changing the alt="CCO" to alt="public domain"
<img src="https://licensebuttons.net/p/zero/1.0/88x31.png"alt="public domain"style="border-style: none;">

2.  Form label
The label should be accurate aand descriptive.
Form label
<label class="overlay overlay-pagetoc" for="__page-toc">
<div class="visually-hidden">
 Table of Contents
</div>
</label>

Structural Elements
1. Unordered list
<ol class="nav bd-sidenav bd-sidenav__home-link">
<li class="toctree-l1">
<a class="reference internal" href="../intro.html">
Project Jupyter Governance
</a>
</li>
</ol>


2. Navigation
A <nav> element or role="navigation" should be present.
<nav>
Unordered list
<ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#the-code-of-conduct-committee">
The Code of Conduct Committee
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#enforcement-guidelines-and-principles">
Enforcement guidelines and principles
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#how-the-committee-will-respond-to-reports">
How the committee will respond to reports
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#incident-response-and-committee-actions">
Incident Response and Committee Actions
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#resolutions">
Resolutions
</a>
</li>
<li class="toc-h2 nav-item toc-entry">
<a class="reference internal nav-link" href="#conflicts-of-interest">
Conflicts of Interest
</a>
</li>
</ul>
</nav>

8 ARIA

1. ARIA

Text appears to be a heading but is not a heading element. Changing it to a heading <h2>

<h2 class="caption">
<span class="caption-text">
Code of Conduct
</span>
</h2>

2. ARIA-LABEL
An aria-labelledby is required in multiple form label
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-labelledby="Toggle navigation sidebar">

<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-labelledby="Toggle in-page Table of Contents">
```

**Comment by preshpi** [comment]
@minrk can you please review this :)

**Comment by minrk** [comment]
Thanks for the suggestions! The [aria-labelled-by](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attribute should contain an _id_ of another element containing the label text. It should not be the text itself.

**Comment by preshpi** [comment]
@minrk Thanks for the correction.
The aria-labelledby text should be placed in another element, and then the ID of that element should be used in the aria-labelledBy attribute.
So I suggest that 

### The HTML to be changed 

```
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
Form label
<label class="overlay overlay-navbar" for="__navigation">
<div class="visually-hidden">
Toggle navigation sidebar
</div>
</label>
```


### The HTML after change

```

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-labelledby="nav-toggle nav-sidebar">
Form label
<label class="overlay overlay-navbar" for="__navigation" id="nav-toggle">
<div class="visually-hidden" id="nav-sidebar">
Toggle navigation sidebar
</div>
</label>
```

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
